easysnmp/interface.c

```diff
@@ -146,8 +146,8 @@ static int __get_type_str(int type, char *str, int log_error);
 static int __get_label_iid(char *name, char **last_label, char **iid,
                            int flag);
 static struct tree *__tag2oid(char *tag, char *iid, oid *oid_arr,
-                              int *oid_arr_len, int *type, int best_guess);
-static int __concat_oid_str(oid *doid_arr, int *doid_arr_len, char *soid_str);
+                              size_t *oid_arr_len, int *type, int best_guess);
+static int __concat_oid_str(oid *doid_arr, size_t *doid_arr_len, char *soid_str);
```

Comment:
If these types are being converted from `int` to `size_t`, there are other parameters in other functions that need to be changed. Example:
```
static int __add_var_val_str(netsnmp_pdu *pdu, oid *name, int name_length,
                             char *val, int len, int type);
```
should be:
```
static int __add_var_val_str(netsnmp_pdu *pdu, oid *name, size_t name_length,
                             char *val, size_t len, int type);
```
I feel like this is going to be like peeling an onion. It would be better to have these changes be merged separately to keep the commit history for the lines clean.

Reply:
I think this is too hard for me. If it doesn't affect the functionality too much then i would prefer leave it as it is. I looked up all type changes which i made, and all dependent code that it affects, and i hope it all has been resolved. Need to add more unit tests for check it more carefully. I i'll try this

Reply:
> I think this is too hard for me. If it doesn't affect the functionality too much then i would prefer leave it as it is. I looked up all type changes which i made, and all dependent code that it affects, and i hope it all has been resolved. Need to add more unit tests for check it more carefully. I i'll try this

Ok. This should probably be a series of follow up commits.